src/figuresClasses.ts

```diff
@@ -1,19 +1,94 @@
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
+
 export interface Figure {
+  shape: Shape;
+  color: Color;
 
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Shape;
 
-}
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    this.shape = 'triangle';
+
+    if (this.isValidTriangle()) {
+      throw new Error('You cannot make a triangle with such sides');
+    }
+  }
+
+  isValidTriangle(): boolean {
```

Comment:
It is a cool method and I like it. Let's make it private, cause you are using it only internally